tests/handlers/test_agent.py

```diff
@@ -30,3 +34,42 @@ def test_post_agent_unauthenticated(test_client, db_session):
 
     assert response.status_code == 403
     assert response.json()['detail'] == "Permission denied."
+
+
+def test_list_agent_empty_list(db_session, test_client, session_cookie):
+    response = test_client.get('/agent', cookies=session_cookie)
+
+    assert response.json() == []
+
+
+def test_list_agent_success(db_session, test_client, session_cookie):
+    agent = Agent(name='test', oauth_client=OAuthClient(type='confidential'))
+    db_session.add(agent)
+    db_session.commit()
+
+    response = test_client.get('/agent', cookies=session_cookie)
+
+    assert response.status_code == 200
+    response_body = response.json()
+
+    assert len(response_body) == 1
+
+    assert response_body[0]['name'] == 'test'
+    # assert response_body[0]['connection_type'] == 'd'
+    oauth_client = response_body[0]['oauth_client']
+    assert oauth_client['agent_id'] is not None
```

Comment:
As mentioned below the oauth client should not be dumped. You can check the `id` on one of the agents in the response list instead.

You can also more accurately assert by storing and comparing the id of the agent you persist.